roles/deploy/tasks/main.yml

```diff
@@ -2,6 +2,24 @@
 - name: Initialize
   deploy_helper: path="{{ project_root }}" state=present
 
+- name: Check whether project source path is a git repo
+  stat: path={{ project_source_path }}/.git
+  register: git_project
+
+- name: Get current git remote URL
+  command: git config --get remote.origin.url
+  args:
+    chdir: "{{ project_source_path }}"
+  register: remote_origin_url
+  when: git_project.stat.exists
+  changed_when: false
+
+- name: Update git remote URL
+  command: git remote set-url origin {{ project_git_repo }}
+  args:
+    chdir: "{{ project_source_path }}"
+  when: remote_origin_url.stdout is defined and remote_origin_url.stdout != project_git_repo
```

Comment:
If the git project doesn't yet exist, "Get current git remote URL" skips, and its registered value is
```
"remote_origin_url": {
  "changed": false,
  "skipped": true
}
```
In this scenario, `remote_origin_url.stdout` is not defined, so without the `defined` check,
`when: remote_origin_url.stdout != project_git_repo`
would cause the task to fail.

My conditional states "if there is a remote url available to compare," but maybe a more intuitive alternative would be "if a git clone/project exists," like this: 
`when: git_project.stat.exists and remote_origin_url.stdout != project_git_repo`.